### PR TITLE
Fargate support (1st iteration)

### PR DESCRIPTION
**Pull request by michaelwittig**
Support for AWS Fargate

- [x] Cluster
- [x] ALB integration
- [x] NLB integration
- [x] Service with single container/task
- [ ] CW Alarms on HealthyHostCount https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html
- [x] Service with two containers/tasks
- [ ] Tests
- [ ] Docs

### CODE REVIEW

File: fargate/load-balancer-alb.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      - !Ref SecurityGroup
+      Subnets: !If
+      - HasLoadBalancerSchemeInternal
+      - !Split [',', {'Fn::ImportValue': !Sub '${ParentVPCStack}-SubnetsPrivate'}]
+      - !Split [',', {'Fn::ImportValue': !Sub '${ParentVPCStack}-SubnetsPublic'}]
+  HttpListener:
+    Type: 'AWS::ElasticLoadBalancingV2::Listener'
+    Properties:
+      DefaultActions:
+      - FixedResponseConfig:
```
Comment (andreaswittig) [inline fargate/load-balancer-alb.yaml]: @michaelwittig What do you think about redirecting HTTP to HTTPS by default?

File: fargate/service.yaml [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+            'awslogs-region': !Ref 'AWS::Region'
+            'awslogs-group': !Ref LogGroup
+            'awslogs-stream-prefix': !Ref 'AWS::StackName'
+      Cpu: !FindInMap [CpuMap, !Ref Cpu, Cpu]
+      ExecutionRoleArn: !GetAtt 'TaskExecutionRole.Arn'
+      Family: !Ref 'AWS::StackName'
+      Memory: !FindInMap [MemoryMap, !Ref Memory, Memory]
+      NetworkMode: awsvpc
+      RequiresCompatibilities: [FARGATE]
+      # TaskRoleArn: TODO how can we suppot a task role?
```
Comment (andreaswittig) [inline fargate/service.yaml]: @michaelwittig What do you think about providing a template for app container + sidecar container (e.g. nginx)?

File: fargate/service.yaml [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+      ComparisonOperator: GreaterThanThreshold
+      Threshold: 0
+      AlarmActions:
+      - {'Fn::ImportValue': !Sub '${ParentAlertStack}-TopicARN'}
+      Dimensions:
+      - Name: LoadBalancer
+        Value: {'Fn::ImportValue': !Sub '${ParentLoadBalancerStack}-LoadBalancerFullName'}
+      - Name: TargetGroup
+        Value: !GetAtt LoadBalancerTargetGroup.TargetGroupFullName
+  LoadBalancerHttpListenerRule:
```
Comment (andreaswittig) [inline fargate/service.yaml]: I'd suggest an alarm on `HealthyHostCount` of the target group as well.

Comment (michaelwittig) [reply fargate/service.yaml]: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html

**Comment by michaelwittig** [comment]
see #249

**Pull request closed without merging**